src/components/settings/SettingsLMSTab/LMSConfigs/utils.tsx

```diff
@@ -0,0 +1,177 @@
+import type { FormFieldValidation } from "../../../forms/FormContext";
+import { BLACKBOARD_OAUTH_REDIRECT_URL, BLACKBOARD_TYPE, CANVAS_OAUTH_REDIRECT_URL, CANVAS_TYPE, INVALID_NAME } from "../../data/constants";
+import handleErrors from "../../utils";
+import { camelCaseDict } from "../../../../utils";
+// @ts-ignore
+import { setWorkflowStateAction, updateFormFieldsAction } from "../../../forms/data/actions.ts";
+import { CanvasConfigCamelCase, CanvasConfigSnakeCase } from "./Canvas/CanvasConfig";
+import { BlackboardConfigCamelCase, BlackboardConfigSnakeCase } from "./Blackboard/BlackboardConfig";
+import { CornerstoneConfigCamelCase, CornerstoneConfigSnakeCase } from "./Cornerstone/CornerstoneConfig";
+import { DegreedConfigCamelCase, DegreedConfigSnakeCase } from "./Degreed/DegreedConfig";
+import { MoodleConfigCamelCase, MoodleConfigSnakeCase } from "./Moodle/MoodleConfig";
+import { SAPConfigCamelCase, SAPConfigSnakeCase } from "./SAP/SAPConfig";
+// @ts-ignore
+import { FormWorkflowErrorHandler, WAITING_FOR_ASYNC_OPERATION } from "../../../forms/FormWorkflow.tsx";
+
+type ConfigCamelCase =
+  BlackboardConfigCamelCase | CanvasConfigCamelCase | CornerstoneConfigCamelCase | DegreedConfigCamelCase | MoodleConfigCamelCase | SAPConfigCamelCase;
+type ConfigSnakeCase =
+  BlackboardConfigSnakeCase | CanvasConfigSnakeCase | CornerstoneConfigSnakeCase | DegreedConfigSnakeCase | MoodleConfigSnakeCase | SAPConfigSnakeCase;
+
+export const LMS_AUTHORIZATION_FAILED = "LMS AUTHORIZATION FAILED";
+
+export async function handleSubmitHelper(
+  enterpriseCustomerUuid: string,
+  transformedConfig: ConfigSnakeCase,
+  existingData: any,
```

Comment:
Should `existingData` be ConfigCamelCase or ConfigSnakeCase?

Reply:
The reason I've left this as any is because since I'm using a dynamic type, it gives me problems when properties (like clientId) don't exist on that type